[charts] Fix ChartsTooltip `container` and `disablePortal` props being ignored (#22690)
mui/mui-x: docs/data/charts/tooltip/tooltip.md

{{"demo": "TooltipPosition.js", "hideToolbar": true, "bg": "playground"}}

## Portaling the tooltip

By default, the tooltip is rendered inside the chart container with a fixed position.
This means plain `overflow: hidden` ancestors don't clip it.

> But an ancestor that establishes a containing block for fixed elements—through `transform`, `filter`, or `will-change`—combined with `overflow: hidden` does clip the tooltip (a common case when a chart sits inside a virtualized list or a transformed cell).
> 
> To render the tooltip outside of such an ancestor, use the `container` prop to provide the element that receives the tooltip—`document.body` in most cases.
> You can pass it through `slotProps.tooltip`, or directly to `ChartsTooltip`/`ChartsTooltipContainer` when composing a custom component.

Author: ```suggestion
By default, the tooltip is rendered inside the chart container with a fixed position.

To render the tooltip outside, use the `container` prop to provide the element that receives the tooltip.
You can pass it through `slotProps.tooltip`, or directly to `ChartsTooltip`/`ChartsTooltipContainer` when composing a custom component.
```
